apps/storage/api/avatar.post.ts

```diff
@@ -14,192 +14,191 @@ import { PutObjectCommand } from '@aws-sdk/client-s3';
 
 //?  Avatar Path: /[type.value]/[userProfilePublicId]/[size]
 
-export default defineEventHandler(async (event) => {
-  const types = [
-    { name: 'user', value: 'u' },
-    { name: 'org', value: 'o' },
-    { name: 'contact', value: 'c' },
-    { name: 'group', value: 'g' }
-  ];
-  const formInputs = await readMultipartFormData(event);
+export default eventHandler({
+  onRequest: [loggedIn],
+  async handler(event) {
+    const types = [
+      { name: 'user', value: 'u' },
+      { name: 'org', value: 'o' },
+      { name: 'contact', value: 'c' },
+      { name: 'group', value: 'g' }
+    ];
+    const formInputs = await readMultipartFormData(event);
 
-  const avatarId = nanoIdLong();
+    const avatarId = nanoIdLong();
 
-  const typeInput = formInputs
-    .find((input) => input.name === 'type')
-    .data.toString('utf8');
+    const typeInput = formInputs
+      .find((input) => input.name === 'type')
+      .data.toString('utf8');
 
-  const typeObject = types.find((t) => t.name === typeInput);
-  if (!typeObject) {
-    setResponseStatus(event, 400);
-    return send(event, 'Missing or invalid type value');
-  }
-
-  const publicIdInput = formInputs.find((input) => input.name === 'publicId');
-  if (!publicIdInput) {
-    setResponseStatus(event, 400);
-    return send(event, 'Missing publicId value');
-  }
-  const publicId = publicIdInput.data.toString('utf8');
-  console.log({ event: event.context.user });
-  const userId = +event.context.user?.id || null;
-  if (!userId) {
-    setResponseStatus(event, 401);
-    return send(event, 'Unauthorized');
-  }
-  if (typeObject.name === 'user') {
-    const profileResponse = await db.query.userProfiles.findFirst({
-      where: eq(userProfiles.publicId, publicId),
-      columns: {
-        id: true,
-        userId: true,
-        avatarId: true
-      }
-    });
-    if (!profileResponse) {
+    const typeObject = types.find((t) => t.name === typeInput);
+    if (!typeObject) {
       setResponseStatus(event, 400);
-      return send(event, 'Invalid user profile ');
+      return send(event, 'Missing or invalid type value');
     }
-    if (+profileResponse.userId !== userId) {
-      setResponseStatus(event, 401);
-      return send(event, 'Unauthorized');
+
+    const publicIdInput = formInputs.find((input) => input.name === 'publicId');
+    if (!publicIdInput) {
+      setResponseStatus(event, 400);
+      return send(event, 'Missing publicId value');
     }
-  } else if (typeObject.name === 'org') {
-    const orgResponse = await db.query.orgs.findFirst({
-      where: eq(orgs.publicId, publicId),
-      columns: {
-        id: true,
-        slug: true,
-        avatarId: true
-      },
-      with: {
-        members: {
-          columns: {
-            userId: true,
-            role: true
-          },
-          where: eq(orgMembers.role, 'admin')
+    const publicId = publicIdInput.data.toString('utf8');
+    const userId = +event.context.user?.id || null;
```

Comment:
Do we need the `+` here?

Reply:
issue with returns sometimes being strings

Reply:
new issue: https://github.com/uninbox/UnInbox/issues/49